### PR TITLE
Improve error handling in VDML

### LINKED ISSUE #117: Improve Vision Sensor error reporting
#### Expected Behavior:
We'd like the vision sensor to have useful error reporting with `errno`

#### Actual Behavior:
We pretty much always return `EINVAL` for the interesting errors that you'd want to know about.

#### Steps to reproduce:
Mess with `vision_read_by_size` and friends

#### System information:
Platform: V5
PROS Kernel Version: 3.1.5

#### Additional Information
https://github.com/purduesigbots/pros/commit/9ca03646e5e98f24bd16910db3ebbd2916a4ee94 has the start of this

### PR DESCRIPTION
**Pull request by edjubuh**
#### Summary:
  - Changed port number validation to return ENXIO when outside the domain of port numbers
  - Changed port validation to return ENODEV when the requested device doesn't match the actual device
  - Vision Sensor read/get functions consistently use EAGAIN when the libv5rts tells us it couldn't get data
  - Remove docstrings for EACCES since it is practically impossible to get that errno (we wait for mutex indefinitely)
  - vision_get_by_sig consistently sets errno to EDOM when unable to find objects that match (valid) arguments

#### Motivation:
Right now, our returned errnos aren't that great.

##### References (optional):
Close #117 

#### Test Plan:
- [X] When objects are in vision sensors FOV, but none are of a particular signature, signature ID is correct and EDOM is errno
```lang=c++
errno = 0; // set to 0 to make sure errno is being set properly... normally wouldn't do this
// errno was 33 b/c there were no objects to see of signature 1
pros::lcd::print(1, "%d (%d)", vision.get_by_sig(0, 1).signature, errno);
errno = 0;
// errno was 0 b/c it worked fine
pros::lcd::print(2, "%d (%d)", vision.get_by_sig(0, 2).signature, errno);
errno = 0;
// errno was flickered between 0/33 b/c there were sometimes objects of 2 to see
pros::lcd::print(3, "%d (%d)", vision.get_by_sig(2, 2).signature, errno);
errno = 0;
// errno was EINVAL (22) because 8 is an invalid signature id
pros::lcd::print(4, "%d (%d)", vision.get_by_sig(8, 2).signature, errno);
errno = 0;
// errno was EADDRINUSE (112) b/c there was a radio plugged in
pros::lcd::print(5, "%d (%d)", pros::c::vision_get_by_sig(21, 0, 1).signature, errno);
errno = 0;
// errno was ENXIO (6) b/c there's no smart port #22
pros::lcd::print(6, "%d (%d)", pros::c::vision_get_by_sig(22, 0, 1).signature, errno);
// errno was ENODEV (19) b/c there's no motor plugged into 2 (or anything for that matter)
errno = 0;
pros::lcd::print(7, "%d (%d)", pros::c::motor_get_voltage(2), errno);
```

### CODE REVIEW

File: src/devices/vdml_vision.c [hunk 1]
```diff
@@ -64,7 +63,7 @@ vision_object_s_t vision_get_by_size(uint8_t port, const uint32_t size_id) {
 	}
 	err = vexDeviceVisionObjectGet(device->device_info, size_id, (V5_DeviceVisionObject*)&rtn);
```
Comment (HotelCalifornia) [inline src/devices/vdml_vision.c]: need to declare `err` here if you're going to remove it above. currently doesn't compile

File: src/devices/vdml_vision.c [hunk 2]
```diff
@@ -151,6 +147,8 @@ int32_t vision_read_by_size(uint8_t port, const uint32_t size_id, const uint32_t
 
 	for (uint32_t i = size_id; i < c; i++) {
 		if (!vexDeviceVisionObjectGet(device->device_info, i, (V5_DeviceVisionObject*)(object_arr + i))) {
+			errno = EAGAIN;
+			object_arr[i] = VISION_OBJECT_ERR_SIG;
```
Comment (HotelCalifornia) [inline src/devices/vdml_vision.c]:
```suggestion
			object_arr[i].signature = VISION_OBJECT_ERR_SIG;
```

File: src/devices/vdml_vision.c [hunk 3]
```diff
@@ -176,15 +174,21 @@ int32_t _vision_read_by_sig(uint8_t port, const uint32_t size_id, const uint32_t
 
 	uint8_t count = 0;
 	for (uint8_t i = 0; i < c; i++) {
-		vexDeviceVisionObjectGet(device->device_info, i, (V5_DeviceVisionObject*)(object_arr + i));
+		if (!vexDeviceVisionObjectGet(device->device_info, i, (V5_DeviceVisionObject*)(object_arr + count))) {
+			errno = EAGAIN;
+			object_arr[i] = VISION_OBJECT_ERR_SIG;
```
Comment (HotelCalifornia) [inline src/devices/vdml_vision.c]:
```suggestion
			object_arr[i].signature = VISION_OBJECT_ERR_SIG;
```

**Comment by HotelCalifornia** [comment]
>We might need to bump the major version number for this if we're going to properly follow semver.

Since major version for us marks the release series, we'd probably bump the minor version instead. That said, I think we've made similar changes in the past and not bumped versions? Unless that was what took us from 3.0 to 3.1 (can't quite remember)

**Comment by baylessj** [comment]
We made breaking changes a few times in the beta phase (between 3.0.0 and 3.1.0) without bumping the minor version number, but as far as I'm aware we have not made any breaking changes since 3.1.0.

**Pull request merged**